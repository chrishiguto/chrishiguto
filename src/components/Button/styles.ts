import styled, { css } from 'styled-components'

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
    background: transparent;
    outline: none;
    border: none;
    cursor: pointer;

    border: 0.1rem solid var(--color-neutral50);
    border-radius: ${theme.border.radius};

    transition: border 0.2s;

    &:hover {
      border: 0.1rem solid var(--color-neutral60);
    }
  `}
`

export const Icon = styled.div`
  > svg {
    color: var(--text-color);
  }
`
