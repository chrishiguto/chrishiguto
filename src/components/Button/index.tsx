import * as S from './styles'

export type ButtonProps = {
  children?: React.ReactNode
  icon?: React.ReactNode
}

const Button = ({ children, icon }: ButtonProps) => (
  <S.Wrapper>
    <S.Icon>{!!icon && icon}</S.Icon>
    {children}
  </S.Wrapper>
)

export default Button
