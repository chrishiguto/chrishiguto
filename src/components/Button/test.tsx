import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import { Github } from '@styled-icons/feather/Github'

import Button from '.'

describe('<Button />', () => {
  it('should render the button', () => {
    const { container } = renderWithTheme(<Button>Button label</Button>)

    expect(
      screen.getByRole('button', { name: /button label/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the button with icon', () => {
    renderWithTheme(
      <Button icon={<Github aria-label="Button icon" />}>Button label</Button>
    )

    expect(screen.getByLabelText(/button icon/i)).toBeInTheDocument()
  })
})
