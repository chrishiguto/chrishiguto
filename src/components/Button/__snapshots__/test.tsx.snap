// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button 1`] = `
.c0 {
  padding: 0.8rem 1.6rem;
  background: transparent;
  outline: none;
  border: none;
  cursor: pointer;
  border: 0.1rem solid var(--border-color);
  border-radius: 0.4rem;
  -webkit-transition: border 0.2s;
  transition: border 0.2s;
}

.c0:hover {
  border: 0.1rem solid var(--border-hover-color);
}

.c1 > svg {
  color: var(--text-color);
}

<button
  class="c0"
>
  <div
    class="c1"
  />
  Button label
</button>
`;
